src/docs/contribute/css.mdx

```diff
@@ -0,0 +1,256 @@
+      subcomponent's name, e.g. `item` or `group`. This enables us to keep
+      related CSS of both the main component and its subcomponents in a single
+      file and see the big picture during development.
+
+   2. When no CSS on the root element is necessary and styling only takes place
+      once a visual modification is invoked by component props, `root` class
+      name can be omitted entirely.
+
+4. **Modifier class names related to the current HTML element must start with
+   `is`** and contain the name of the target element, e.g. `isRootLoading`
+   (modifies `root`) or `isLabelHidden` (modifies `label`). However, child
+   elements may be modified as well, e.g. by a CSS selector like
+   `.isRootRequired > .label` (root is marked as required but the label is what
+   needs to be visually modified).
+
+5. **Modifier class names related to child elements must start with `has`**
+    and refer to the element in question, e.g. `hasRootSmallInput` (applies
+    styling on `root` but relates to `input`).
+
+## Custom Properties
+
+React UI takes advantage of supporting [modern browsers] and uses
+[CSS custom properties] to make writing and maintaining CSS more efficient.
+
+There are three kinds of custom properties used:
+
+1. `--rui-local-*` for internal (component-scoped, local) custom properties.
+   May reuse other custom property types.
+
+2. `--rui-custom-*` for any custom properties whose value comes from component's
+   API. May reuse other custom property types.
+
+3. `--rui-*` (unscoped for the sake of brevity) for theme-related custom
+   properties. Part of public API, designed to be customized. Must not reuse
+   other custom property types. Refer to [theming overview] to learn how their
+   names are created.
+
+## Preprocessing with Sass
+
+All React UI CSS source is written in SCSS syntax of [Sass] preprocessor.
+
+- **Sass variables, mixins and functions must use kebab-case notation.**
+
+- **Only [Sass modules] must be used to organize** Sass source files, `@import`
+  is deprecated. Using scoped variables, mixins and functions (those starting
+  with `_`) is highly recommended whenever appropriate.
+
+- **[Built-in Sass modules] should be preferred** over older Sass functions that
+  are deprecated, e.g. `map.get()` instead of `map-get()`.
+
+- **Mixins that lead to duplicate CSS should be avoided.** If possible, combine
+  multiple CSS selectors for the desired rule set to achieve the same result.
+
+- **Extend functionality should be avoided entirely** due to its hardly
+  predictable behavior.
+
+- **Classes that are automatically generated by Sass loops should be handled
+  with care.** With loops, it's easy to produce a lot of CSS and negatively
+  impact performance.
+
+## Postprocessing with PostCSS
+
+All styles are automatically prefixed by [Autoprefixer] plugin for [PostCSS]
+according to [Browserslist] configuration stored in `.browserslistrc`.
+
+[Stylelint]: https://stylelint.io
```

Comment:
These links do not get rendered in final docs.

Also why are they here? I think we should make a hyperlink out of name used in text, but this is just a bunch of more or less related texts. 

It makes it feel like a CSS tutorial and not a library documentation.

Reply:
They do render, it's called [reference links](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links).

Reply:
I dont see them:
![Screenshot from 2021-09-16 10-53-08](https://user-images.githubusercontent.com/4589176/133582144-b5e40563-e959-41fb-b5e2-bf171fbfdcea.png)

Reply:
@mbohal It is not supposed to be rendered, see the link. It is just definition so you do not have to have long links in the main texts and you use those predefined instead.

Reply:
Everyday I learn something new…